lib/module.js

```diff
@@ -169,6 +169,9 @@ Module._findPath = function(request, paths) {
     }
 
     if (filename) {
+      if (request === '.' && i > 0) {
+        console.error(`(node) warning: require('.') resolved to ${filename}`);
+      }
```

Comment:
I'd be fine with something like that, but in the long term, we probably need something like `util.warnOnce(msg)`. These probably shouldn't be public methods either.